tests/unit_tests/test_routes.py

```diff
@@ -100,3 +100,94 @@ def test_invalid_kml_post(self):
+            location = {'LocationConstraint': AWS_S3_REGION_NAME}
+            s3bucket.create_bucket(Bucket=AWS_S3_BUCKET_NAME, CreateBucketConfiguration=location)
+        except s3bucket.meta.client.exceptions.BucketAlreadyExists as err:
+            logger.debug(
+                "Bucket %s already exists but should not.", err.response['Error']['BucketName']
+            )
+            raise err
+
+        try:
+            dynamodb = boto3.resource(
+                'dynamodb', region_name=AWS_DB_REGION_NAME, endpoint_url=AWS_DB_ENDPOINT_URL
+            )
+            dynamodb.create_table(
+                TableName=AWS_DB_TABLE_NAME,
+                AttributeDefinitions=[
+                    {
+                        'AttributeName': 'admin_id', 'AttributeType': 'S'
+                    },
+                ],
+                KeySchema=[
+                    {
+                        'AttributeName': 'admin_id', 'KeyType': 'HASH'
+                    },
+                ]
+            )
+        except dynamodb.meta.client.exceptions.ResourceInUseException as err:
+            logger.debug("Table %s already exists but should not.", AWS_DB_TABLE_NAME)
+            raise err
+
+    def tearDown(self):
+        s3bucket = boto3.resource('s3', region_name=AWS_S3_REGION_NAME).Bucket(AWS_S3_BUCKET_NAME)
+        for key in s3bucket.objects.all():
+            key.delete()
+        s3bucket.delete()
+        dynamodb = boto3.resource(
+            'dynamodb', region_name=AWS_DB_REGION_NAME, endpoint_url=AWS_DB_ENDPOINT_URL
+        )
+        dynamodb.Table(AWS_DB_TABLE_NAME).delete()
+
+    def test_get_id(self):
+        # first step: create a kml file to retrieve
+        response = self.app.post(
+            "/kml", data=self.kml_string, content_type="application/vnd.google-earth.kml+xml"
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.content_type, "application/json")
+
+        # second step : fetch the id
+
+        id_to_fetch = response.json['id']
+        stored_geoadmin_link = response.json['links']['kml']
+        stored_kml_admin_link = response.json['links']['self']
+
+        # third step : test the get id endpoint
+
+        response = self.app.get(f"/kml/{id_to_fetch}")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content_type, "application/json")
+
+        self.assertEqual(stored_geoadmin_link, response.json['links']['kml'])
+        self.assertEqual(stored_kml_admin_link, response.json['links']['self'])
+
+    def test_get_id_nonexistent(self):
+        id_to_fetch = 'nonExistentId'
+        response = self.app.get(f"kml/{id_to_fetch}")
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.content_type, "application/json")
+        self.assertEqual(
+            response.json['error']['message'], f'{id_to_fetch} is not an existing kml id.'
+        )
```

Comment:
Maybe integrate that in the postendpoint test, that the test is not so long and not repeating. What do you think?

Reply:
That could be a good idea not to copy code, however it's not a good idea to define get endpoint tests in a post endpoint test class. 

A better way would be to create a base class, then use this base class to create the setups and teardowns.

@ltshb what do you think, should I create a ticket and tacle this at the same time ?

Reply:
@ltkum no need for a ticket but go ahead for the implementation in a base class.

Reply:
It's done